### ISSUE TITLE
Relayer related crash

### ISSUE THREAD

**Issue by faddat** [cosmos/gaia#930]
## Summary of Bug

```
        github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecbe0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000040001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecb40, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000030001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecaa0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x7f0f00020001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346eca00, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x10001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc020392640, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*ImmutableTree).IterateRange(0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1f42101, 0xc019009fb8, 0x212d280)
        github.com/cosmos/iavl@v0.16.0/immutable_tree.go:184 +0xd2
github.com/cosmos/cosmos-sdk/store/iavl.(*iavlIterator).iterateRoutine(0xc01d712fc0)
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:422 +0x98
created by github.com/cosmos/cosmos-sdk/store/iavl.newIAVLIterator
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:415 +0x18c
[root@notional .gaia]#
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecc80, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x50001, 0xc0122c1f48, ...)
                             github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecbe0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000040001, 0xc0122c1f48, ...)
                             github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecb40, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000030001, 0xc0122c1f48, ...)
                             github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecaa0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x7f0f00020001, 0xc0122c1f48, ...)
                             github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346eca00, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x10001, 0xc0122c1f48, ...)
                             github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc020392640, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1, 0xc01github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*ImmutableTree).IterateRange(0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1f42101, 0xc019009fb8, 0x212d280)
                             github.com/cosmos/iavl@v0.16.0/immutable_tree.go:184 +0xd2
github.com/cosmos/cosmos-sdk/store/iavl.(*iavlIterator).iterateRoutine(0xc01d712fc0)
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:422 +0x98
created by github.com/cosmos/cosmos-sdk/store/iavl.newIAVLIterator
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:415 +0x18c
50001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecbe0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000040001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecb40, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000030001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecaa0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x7f0f00020001, 0xc0122c1f48, ...)
      github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346eca00, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x10001, 0xc0122c1f48, ...)
      github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc020392640, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1, 0xc0122c1f48, ...)
      github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*ImmutableTree).IterateRange(0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1f42101, 0xc019009fb8, 0x212d280)
      github.com/cosmos/iavl@v0.16.0/immutable_tree.go:184 +0xd2
github.com/cosmos/cosmos-sdk/store/iavl.(*iavlIterator).iterateRoutine(0xc01d712fc0)
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:422 +0x98
created by github.com/cosmos/cosmos-sdk/store/iavl.newIAVLIterator
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:415 +0x18c
[root@notional .gaia]#
1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecbe0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000040001, 0xc0122c1f48, ...)
                                                                                                                                                        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecb40, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000030001, 0xc0122c1f48, ...)
                                                                                                                                                        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecaa0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x7f0f00020001, 0xc0122c1f48, ...)                                                                                                                                                       github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346eca00, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x10001, 0xc0122c1f48, ...)
                                                                                                                                                        github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc020392640, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1, 0xc0122c1f48, ...)
                                                                                                                                                        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*ImmutableTree).IterateRange(0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1f42101, 0xc019009fb8, 0x212d280)
                                                                                                                                                        github.com/cosmos/iavl@v0.16.0/immutable_tree.go:184 +0xd2
github.com/cosmos/cosmos-sdk/store/iavl.(*iavlIterator).iterateRoutine(0xc01d712fc0)
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:422 +0x98
created by github.com/cosmos/cosmos-sdk/store/iavl.newIAVLIterator
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:415 +0x18c
[root@notional .gaia]#
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecdc0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x400070001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecd20, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000060001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecc80, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x50001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecbe0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000040001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecb40, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0xc000030001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346ecaa0, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x7f0f00020001, 0xc0122c1f48, ...)
github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*Node).traverseInRange(0xc0346eca00, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x10001, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:485 +0x2be
github.com/cosmos/iavl.(*Node).traverseInRange(0xc020392640, 0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1, 0xc0122c1f48, ...)
        github.com/cosmos/iavl@v0.16.0/node.go:479 +0x1e9
github.com/cosmos/iavl.(*ImmutableTree).IterateRange(0xc01294c9c0, 0xc032330f60, 0x2a, 0x2a, 0xc032330f90, 0x2a, 0x2a, 0x1f42101, 0xc019009fb8, 0x212d280)
        github.com/cosmos/iavl@v0.16.0/immutable_tree.go:184 +0xd2
github.com/cosmos/cosmos-sdk/store/iavl.(*iavlIterator).iterateRoutine(0xc01d712fc0)
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:422 +0x98
created by github.com/cosmos/cosmos-sdk/store/iavl.newIAVLIterator
        github.com/cosmos/cosmos-sdk@v0.42.6/store/iavl/store.go:415 +0x18c
   
```
## Version

5.0.4

## Steps to Reproduce

I am not sure that this will work to reproduce but am certain it happened when I did a `systemctl restart hermes`.
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by yaruwangway** [comment]
Hi @crodriguezvega, do you have any ideas about this ? would this still be an issue in the new version ?

**Comment by faddat** [comment]
hi, this shouldn't be an issue in the new version and went away completely when I began to use Tendermint v0.34.14 in my custom builds of gaia for relaying, which are here:

https://github.com/notional-labs/gaia

I consider this one closed, and posit that the issue was with v0.34.13
